scripts/partition_manager_report.py

```diff
@@ -48,6 +48,16 @@ def print_region(region, size, pm_config):
                 hex(pm_config[name]['size']),
                 bcolors['ENDC'])
             for name in sorted_pm_config]
+    endspan = 0
+    i = 0
+    for name in sorted_pm_config[:-1]:
```

Comment:
This is a second loop of `name in sorted_pm_config`, the first loop we have in [L50](https://github.com/nrfconnect/sdk-nrf/blob/1b3c72a5ca68628694c5c9df9f176bea1004ba37/scripts/partition_manager_report.py#L50)

If refactoring the code a little, then there could be just a single `name in sorted_pm_config` loop, and you could avoid both the `i = 0`, `i += 1`, and `lines.insert` code lines, as you would be able add the separator line `------` when a span ends, instead of doing a second loop and then having to do an insert into existing array.

This would be a code cleanup that I think would be nice, but not a blocker for approving this PR.